contracts/vesting/src/contract.rs

```diff
@@ -156,28 +173,79 @@ pub fn claim(deps: DepsMut, env: Env, info: MessageInfo) -> StdResult<Response>
 }
 
 fn compute_claim_amount(current_time: u64, vesting_info: &VestingInfo) -> Uint128 {
-    let mut claimable_amount: Uint128 = Uint128::zero();
-    for s in vesting_info.schedules.iter() {
-        if s.0 > current_time || s.1 < vesting_info.last_claim_time {
-            continue;
-        }
+    let last_claim_time = vesting_info.last_claim_time;
+    vesting_info
+        .schedules
+        .iter()
+        .filter(|vs| vs.start_time < current_time && vs.end_time > last_claim_time)
+        .map(|vs| {
+            let passed_time = min(vs.end_time, current_time) - max(vs.start_time, last_claim_time);
+
+            // prevent zero time_period case
+            let time_period = vs.end_time - vs.start_time;
+            let release_amount_per_time = Decimal::from_ratio(vs.amount, time_period);
+
+            Uint128::from(passed_time as u128) * release_amount_per_time
+        })
+        .fold(Uint128::zero(), |acc, i| acc + i)
+}
 
-        // min(s.1, current_time) - max(s.0, last_claim_time)
-        let passed_time =
-            std::cmp::min(s.1, current_time) - std::cmp::max(s.0, vesting_info.last_claim_time);
+fn compute_claim_amounts(
+    current_time: u64,
+    vesting_infos: &[(CanonicalAddr, VestingInfo)],
+) -> Uint128 {
+    vesting_infos
+        .iter()
+        .map(|p| &p.1)
+        .map(|vi| compute_claim_amount(current_time, vi))
+        .fold(Uint128::zero(), |acc, i| acc + i)
+}
 
-        // prevent zero time_period case
-        let time_period = s.1 - s.0;
-        let release_amount_per_time: Decimal = Decimal::from_ratio(s.2, time_period);
+pub fn withdraw_unclaimed(deps: DepsMut, env: Env, info: MessageInfo) -> ContractResult<Response> {
+    let current_time = env.block.time.seconds();
+    let config = read_config(deps.storage)?;
+    let address = info.sender.to_string();
 
-        claimable_amount += Uint128::from(passed_time as u128) * release_amount_per_time;
+    let last_claim_deadline = config.last_claim_deadline;
+    if last_claim_deadline > current_time {
+        return Err(ContractError::LastClaimDeadlineNotExceeded);
     }
 
-    claimable_amount
+    let vesting_infos = read_vesting_infos(deps.storage, None, None, None)?;
+    let unclaimed_amount = compute_claim_amounts(current_time, &vesting_infos);
+
+    let messages: Vec<CosmosMsg> = if unclaimed_amount.is_zero() {
+        vec![]
+    } else {
+        vec![CosmosMsg::Wasm(WasmMsg::Execute {
+            contract_addr: deps.api.addr_humanize(&config.anchor_token)?.to_string(),
+            funds: vec![],
+            msg: to_binary(&Cw20ExecuteMsg::Transfer {
+                recipient: address.clone(),
```

Comment:
Wasn't sure about this. Right now, this can only be called by the `owner`. Should it also be configurable?